services/wallet/router/router_v2.go

```diff
@@ -294,98 +322,141 @@ func findBestV2(routes [][]*PathV2, tokenPrice float64, nativeChainTokenPrice fl
 func validateInputData(input *RouteInputParams) error {
 	if input.SendType == ENSRegister {
 		if input.Username == "" || input.PublicKey == "" {
-			return errors.New("username and public key are required for ENSRegister")
+			return routererrors.ErrUsernameAndPubKeyRequiredForENSRegister
 		}
 		if input.TestnetMode {
 			if input.TokenID != pathprocessor.SttSymbol {
-				return errors.New("only STT is supported for ENSRegister on testnet")
+				return routererrors.ErrOnlySTTSupportedForENSRegisterOnTestnet
 			}
 		} else {
 			if input.TokenID != pathprocessor.SntSymbol {
-				return errors.New("only SNT is supported for ENSRegister")
+				return routererrors.ErrOnlySTTSupportedForENSReleaseOnTestnet
 			}
 		}
 		return nil
 	}
 
 	if input.SendType == ENSRelease {
 		if input.Username == "" {
-			return errors.New("username is required for ENSRelease")
+			return routererrors.ErrUsernameRequiredForENSRelease
 		}
 	}
 
 	if input.SendType == ENSSetPubKey {
 		if input.Username == "" || input.PublicKey == "" || ens.ValidateENSUsername(input.Username) != nil {
-			return errors.New("username and public key are required for ENSSetPubKey")
+			return routererrors.ErrUsernameAndPubKeyRequiredForENSSetPubKey
 		}
 	}
 
 	if input.SendType == StickersBuy {
 		if input.PackID == nil {
-			return errors.New("packID is required for StickersBuy")
+			return routererrors.ErrPackIDRequiredForStickersBuy
 		}
 	}
 
 	if input.SendType == Swap {
 		if input.ToTokenID == "" {
-			return errors.New("toTokenID is required for Swap")
+			return routererrors.ErrToTokenIDRequiredForSwap
 		}
 		if input.TokenID == input.ToTokenID {
-			return errors.New("tokenID and toTokenID must be different")
+			return routererrors.ErrTokenIDAndToTokenIDDifferent
 		}
 
 		// we can do this check, cause AmountIn is required in `RouteInputParams`
 		if input.AmountIn.ToInt().Cmp(pathprocessor.ZeroBigIntValue) > 0 &&
 			input.AmountOut != nil &&
 			input.AmountOut.ToInt().Cmp(pathprocessor.ZeroBigIntValue) > 0 {
-			return errors.New("only one of amountIn or amountOut can be set")
+			return routererrors.ErrOnlyOneOfAmountInOrOutSet
 		}
 
 		if input.AmountIn.ToInt().Sign() < 0 {
-			return errors.New("amountIn must be positive")
+			return routererrors.ErrAmountInMustBePositive
 		}
 
 		if input.AmountOut != nil && input.AmountOut.ToInt().Sign() < 0 {
-			return errors.New("amountOut must be positive")
+			return routererrors.ErrAmountOutMustBePositive
 		}
 	}
 
 	if input.FromLockedAmount != nil && len(input.FromLockedAmount) > 0 {
+		copyMap := func(src map[uint64]bool) map[uint64]bool {
```

Comment:
maybe make it a real func rather than anonymous?

Reply:
For now it's used only here, we can move it out when there is a need to have/use it in another place. 
I was thinking of moving it to common.go, but it's not that general, cause it works with certain type only `map[uint64]bool`.

Reply:
There is actually a function for this in `filter.go`